### PR TITLE
add tool interfaces

### PR DESCRIPTION
**Pull request by prayansh**

### CODE REVIEW

**Comment by Wundero** [comment]
Also a quick question on the 'edit' message: Is this a complete overwrite of the previously existing feature, or is it an addition style modification (for example, PenTool edit could be just 1 or 2 more points on the end, and the packet sends just those 2 points, or it sends the original points + those 2). The overwrite would be much easier to implement but it is important to know for when we go to do so.

**Comment by prayansh** [comment]
> Also a quick question on the 'edit' message: Is this a complete overwrite of the previously existing feature, or is it an addition style modification (for example, PenTool edit could be just 1 or 2 more points on the end, and the packet sends just those 2 points, or it sends the original points + those 2). The overwrite would be much easier to implement but it is important to know for when we go to do so.

I think its an implementation choice, Im ok with either, so will leave upto u to decide.

File: client/src/utils/types.ts [hunk 1]
```diff
@@ -26,4 +36,77 @@ export interface JoinRoomMessage {
...
+
+type Color = string
+
+type Point = {
+  x: number
+  y: number
+}
+
+interface Drawable {
+  origin: Point
```
Comment (Wundero) [inline client/src/utils/types.ts:60]: Why abstract 'origin' only to include it in the subtypes?

File: client/src/utils/types.ts [hunk 2]
```diff
@@ -26,4 +36,77 @@ export interface JoinRoomMessage {
...
+
+export interface PenTool extends Drawable, Draggable {
+  type: typeof TOOL_PEN
+  points: Point[] // all except origin point
+}
+
+export interface TextTool extends Drawable, Draggable, Rotatable {
+  type: typeof TOOL_TEXT
+  text: string
+  origin: Point // Top-left
```
Comment (Wundero) [inline client/src/utils/types.ts:88]: What is the purpose of this origin if it extends a type with an origin?

**Pull request merged**